### PR TITLE
PRIME 1721 gis updates

### PR DESCRIPTION
**Pull request by Paahn**
- PRIME 1721 + 1784

### CODE REVIEW

File: prime-dotnet-webapi/HttpClients/LDAP Api/LdapClient.cs [hunk 1]
```diff
@@ -42,7 +43,13 @@ public async Task<string> GetUserAsync(string username, string password)
                 if (response.IsSuccessStatusCode)
                 {
                     var successResponse = JsonConvert.DeserializeObject<GisUser>(responseJsonString);
```
Comment (james-hollinger) [inline prime-dotnet-webapi/HttpClients/LDAP Api/LdapClient.cs:45]: Why de-serialize to `GisUser` then map to `LdapResponseKeys` when you can just de-serialize to `LdapResponseKeys` or return the `GisUser`?

File: prime-dotnet-webapi/Services/GisService.cs [hunk 2]
```diff
@@ -83,10 +85,17 @@ public async Task<int> CreateOrUpdateGisEnrolmentAsync(GisChangeModel changeMode
             return currentGisEnrolment.Id;
         }
 
-        public async Task<bool> LdapLogin(string username, string password, ClaimsPrincipal user)
+        public async Task<LoginStatus> LdapLogin(string username, string password, ClaimsPrincipal user)
         {
-            var gisUserRole = await _ldapClient.GetUserAsync(username, password);
-            var success = gisUserRole == "GISUSER";
+            var ldapResponse = await _ldapClient.GetUserAsync(username, password);
+            var success = ldapResponse.GisUserRole == "GISUSER";
```
Comment (james-hollinger) [inline prime-dotnet-webapi/Services/GisService.cs:91]: Now that we are looking at the LDAP API more thoroughly, is `UserRole == "GISUSER"` really all that is required for the login to be successful? the LDAP API returns an `Authenticated` property, should we be checking that?

File: prime-angular-frontend/src/app/modules/gis-enrolment/pages/ldap-information-page/ldap-information-page.component.ts [hunk 3]
```diff
@@ -1,9 +1,9 @@
 import { Component, OnInit } from '@angular/core';
-import { FormControl } from '@angular/forms';
 import { ActivatedRoute, Router } from '@angular/router';
 import { MatDialog } from '@angular/material/dialog';
 
-import { exhaustMap } from 'rxjs/operators';
+import { EMPTY } from 'rxjs';
+import { exhaustMap, map } from 'rxjs/operators';
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/modules/gis-enrolment/pages/ldap-information-page/ldap-information-page.component.ts:6]: seems map is imported but not used

File: prime-angular-frontend/src/app/modules/gis-enrolment/shared/resources/gis-enrolment-resource.service.ts [hunk 4]
```diff
@@ -1,4 +1,5 @@
 import { Injectable } from '@angular/core';
+import { HttpErrorResponse } from '@angular/common/http';
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/modules/gis-enrolment/shared/resources/gis-enrolment-resource.service.ts:2]: imported but not used

File: prime-angular-frontend/src/app/modules/gis-enrolment/pages/ldap-information-page/ldap-information-page.component.html [hunk 5]
```diff
@@ -39,12 +39,26 @@
...
+              will need to wait {{ lockoutTimeInHours }} hour(s) before trying again. If you need assistance please
+              contact Reg Admin at (999) 999-9999.
+            </ng-container>
+
+            <ng-container #alertContent
+                          *ngIf="!remainingAttempts"
+                          class="alert-content">
+              Your account is now locked due to too many failed login attempts. Please wait {{ lockoutTimeInHours }}
+              hour(s) before trying again. If you think this is a mistake please contact Reg Admin at (999) 999-9999.
+            </ng-container>
```
Comment (johnnyl6) [inline prime-angular-frontend/src/app/modules/gis-enrolment/pages/ldap-information-page/ldap-information-page.component.html:59]: wonder if below sample would reduce the redundancy a little bit.
```html
<ng-container #alertContent class="alert-content">
  <ng-container *ngIf="remainingAttempts === 1; else #lockedBlock">
     You have had multiple failed login attempts the next failed attempt will lock your account, and you
     will need to wait {{ lockoutTimeInHours }} hour(s) before trying again. If you need assistance please
     contact Reg Admin at (999) 999-9999.
  </ng-container>
  <ng-template #lockedBlock>
     Your account is now locked due to too many failed login attempts. Please wait {{ lockoutTimeInHours }}
     hour(s) before trying again. If you think this is a mistake please contact Reg Admin at (999) 999-9999.
  </ng-template>
<ng-container>
```

File: prime-dotnet-webapi/Controllers/GisController.cs [hunk 6]
```diff
@@ -4,6 +4,7 @@
 
 using Prime.Auth;
 using Prime.Services;
+using Prime.HttpClients;
```
Comment (johnnyl6) [inline prime-dotnet-webapi/Controllers/GisController.cs:7]: unused using directive

File: prime-angular-frontend/src/app/modules/gis-enrolment/pages/ldap-information-page/ldap-information-page.component.html (hunk 5 above)
Comment (mtpultz) [reply prime-angular-frontend/src/app/modules/gis-enrolment/pages/ldap-information-page/ldap-information-page.component.html:59]: I had something similar to Johnny's suggestion in that example branch did that not work?

Comment (mtpultz) [reply prime-angular-frontend/src/app/modules/gis-enrolment/pages/ldap-information-page/ldap-information-page.component.html:59]: The first `*ngIf` it should filter out anything that is greater than 1

```
<ng-container *ngIf="hasAttemptedSubmission && remainingAttempts <= 1 && lockoutTimeInHours">
```

**Review by mtpultz — APPROVED** [review]
Looks good.  We'll want to merge this in after the remaining updates so we can quickly test everything in DEV tomorrow (Friday) and have it pushed to TEST so it's ready for Tuesday.

**Review by AMiKaH — APPROVED** [review]

**Review by james-hollinger — APPROVED** [review]
LGTM

**Pull request merged**